py/torch_tensorrt/fx/converters/converter_utils.py

```diff
@@ -151,28 +151,50 @@ def extend_mod_attr_to_tuple(mod: torch.nn.Module, name: str, size: int):
     return extend_attr_to_tuple(val, size)
 
 
-def to_numpy(tensor: Optional[torch.Tensor]) -> Optional[np.ndarray]:
+def to_numpy(
+    value: Optional[Union[torch.Tensor, np.ndarray, int, float]],
+    dtype: Optional[Union[torch.dtype, np.dtype, TRTDataType]] = None,
+) -> Optional[np.ndarray]:
     """
-    Convert a PyTorch Tensor to a Numpy Array. If the tensor is
+    Convert a PyTorch Tensor, Numpy array, or scalar to a Numpy Array. If the tensor is
     quantized it will be dequantized first.
 
     Args:
-        tensor (Optional[torch.Tensor]): A PyTorch tensor or None.
+        value (Optional[Union[torch.Tensor, np.ndarray, int, float]]):
+            A PyTorch tensor, Numpy array, int, or float
+        dtype (Optional[Union[torch.dtype, np.dtype, TRTDataType]]):
+            If a dtype is given, we will convert the type of the given `value` to this dtype.
 
     Returns:
         A Numpy array.
     """
+    output = None
 
-    if tensor is None:
-        return tensor
+    if value is None or isinstance(value, np.ndarray):
+        output = value
 
-    assert isinstance(
-        tensor, torch.Tensor
-    ), f"to_numpy can only be called on None or a torch.Tensor, got: {tensor}"
-    if tensor.is_quantized:
-        tensor = tensor.dequantize()
+    elif isinstance(value, torch.Tensor):
+        if value.is_quantized:
+            value = value.dequantize()
 
-    return tensor.cpu().detach().contiguous().numpy()
+        output = value.cpu().detach().contiguous().numpy()
+
+    elif isinstance(value, int):
+        output = np.array([value], dtype=np.int32)
+
+    elif isinstance(value, float):
+        output = np.array([value], dtype=np.float32)
```

Comment:
add support for fp16?

Reply:
For this particular instance, the intent was just to take Python builtins and translate them to numpy arrays. If the caller specifies an FP16 dtype in the function schema, that cast will be applied later, on line 196.